From f125c97ec5fda2b3e4fb69868eebd521ac90e366 Mon Sep 17 00:00:00 2001
From: Aleksander Morgado <aleksander@aleksander.es>
Date: Thu, 26 Nov 2020 23:30:56 +0100
Subject: [PATCH] broadband-modem-mbim: plug memleak when processing USSD in
 GSM7

(cherry picked from commit 0729ba5c88eb038e60ffbe3faf9431e9650f7ca1)
---
 src/mm-broadband-modem-mbim.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/mm-broadband-modem-mbim.c b/src/mm-broadband-modem-mbim.c
index 2ff3a0b4..c93b2e2d 100644
--- a/src/mm-broadband-modem-mbim.c
+++ b/src/mm-broadband-modem-mbim.c
@@ -4612,8 +4612,8 @@ ussd_decode (guint32      scheme,
     gchar *decoded = NULL;
 
     if (scheme == MM_MODEM_GSM_USSD_SCHEME_7BIT) {
-        guint8  *unpacked;
-        guint32  unpacked_len;
+        g_autofree guint8  *unpacked = NULL;
+        guint32             unpacked_len;
 
         unpacked = mm_charset_gsm_unpack ((const guint8 *)data->data, (data->len * 8) / 7, 0, &unpacked_len);
         decoded = (gchar *) mm_charset_gsm_unpacked_to_utf8 (unpacked, unpacked_len);
-- 
2.25.1

